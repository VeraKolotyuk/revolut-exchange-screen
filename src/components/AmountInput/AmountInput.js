import IMask from 'imask';
import {useRef, useEffect} from 'react';
import './style.css';

const AmountInput = (props) => {
    const input = useRef(null);
    useEffect(() => {
        IMask(input.current, {
            mask: Number,  // enable number mask
            scale: 2,  // digits after point, 0 for integers
            signed: true,  // disallow negative
            thousandsSeparator: '',  // any single char
            padFractionalZeros: false,  // if true, then pads zeros at end to the length of scale
            normalizeZeros: true,  // appends or removes zeros at ends
            radix: '.',  // fractional delimiter
            mapToRadix: ['.']
        });
    },[])

    return (
        <div>
            <input
                ref={input}
                type="text"
                value={props.value}
                onChange={props.onInputChangeHandler}
                placeholder={"0"}
                className="exchange-amount-input"
            />
            {props.showError && <div className="form-error">exceeds balance</div>}
        </div>
    )
}

export default AmountInput;