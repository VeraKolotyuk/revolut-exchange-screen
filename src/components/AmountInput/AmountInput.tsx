import {IMaskInput} from 'react-imask';
import './style.css';
import {EXCEEDS_BALANCE_MESSAGE, EXCHANGE_INPUT_SCALE} from '../../utils/constants';

type Props = {
    value: number,
    showError: boolean,
    onInputChangeHandler: (a: { target: HTMLInputElement;}) => void,
};

const AmountInput = ({value, onInputChangeHandler, showError }: Props) => {
    const inputIMaskProps = {
        mask: Number,
        scale: EXCHANGE_INPUT_SCALE,
        signed: true,
        thousandsSeparator: '',
        padFractionalZeros: false,
        normalizeZeros: true,
        radix: '.',
        mapToRadix: ['.']
    };

    return (
        <div>
            <IMaskInput
                {...inputIMaskProps}
                type='text'
                value={value.toString()}
                onChange={onInputChangeHandler}
                placeholder={'0'}
                className="exchange-amount-input"
            />
            {showError && <div className="form-error">{EXCEEDS_BALANCE_MESSAGE}</div>}
        </div>
    );
};

export default AmountInput;