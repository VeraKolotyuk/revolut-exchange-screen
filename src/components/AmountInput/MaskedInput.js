import {IMaskInput} from 'react-imask';
import {EXCHANGE_INPUT_SCALE} from '../../utils/constants';

const MaskedInput = ({value, onInputChangeHandler}) => {
    const inputIMaskProps = {
        mask: Number,
        scale: EXCHANGE_INPUT_SCALE,
        signed: true,
        thousandsSeparator: '',
        padFractionalZeros: false,
        normalizeZeros: true,
        radix: '.',
        mapToRadix: ['.']
    };
    return (
        <IMaskInput
            {...inputIMaskProps}
            type='text'
            value={value.toString()}
            onChange={onInputChangeHandler}
            placeholder={'0'}
            className="exchange-amount-input"
        />
    );
};

export default MaskedInput;