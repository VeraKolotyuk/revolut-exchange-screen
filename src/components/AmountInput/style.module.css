.exchange-amount-input {
    border: none;
    outline: none;
    text-align: right;
    font-weight: 700;
    font-size: 1.1rem;
}

.form-error {
    color: firebrick;
    font-size: .8rem;
    text-align: right;
    font-weight: 400;
}