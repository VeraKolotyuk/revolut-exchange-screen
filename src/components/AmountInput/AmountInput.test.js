import AmountInput from './AmountInput';
import { shallow, configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

configure({adapter: new Adapter()});


describe('AmountInput rendering', () => {
    it('renders correctly', () => {
        let wrapper = shallow(<Provider store={mockStore()}>
            <AmountInput value={10.23} onInputChangeHandler={jest.fn()} showError={false} />
        </Provider>);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with error message', () => {
        let wrapper = mount(<AmountInput value={10.23} onInputChangeHandler={jest.fn()} showError={true} />);
        expect(wrapper.find('.form-error').exists()).toBeTruthy();
    });
});