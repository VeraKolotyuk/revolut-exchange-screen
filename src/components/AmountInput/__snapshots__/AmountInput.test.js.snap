// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountInput rendering renders correctly 1`] = `ShallowWrapper {}`;

exports[`AmountInput rendering renders correctly with error message 1`] = `ShallowWrapper {}`;
